pkg/envoy/lds/builder_test.go

```diff
@@ -0,0 +1,336 @@
+package lds
+
+import (
+	"testing"
+
+	xds_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	xds_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	xds_tcp_proxy "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/tcp_proxy/v3"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/rds/route"
+	"github.com/openservicemesh/osm/pkg/identity"
+	"github.com/openservicemesh/osm/pkg/service"
+	"github.com/openservicemesh/osm/pkg/tests"
+	"github.com/openservicemesh/osm/pkg/trafficpolicy"
+)
+
+func TestHTTPConnManagerBuilder(t *testing.T) {
+	contains := func(filters []*xds_hcm.HttpFilter, filterName string) bool {
+		for _, f := range filters {
+			if f.Name == filterName {
+				return true
+			}
+		}
+		return false
+	}
+
+	testCases := []struct {
+		name       string
+		buildFunc  func(b *httpConnManagerBuilder)
```

Comment:
Alternatively, you could simply pass in the builder object with the build data supplied, although this doesn't test any potential logic in the builder's Setter methods

ie:

```
testCases := []struct {
		name       string
		buildFunc  b *httpConnManagerBuilder
		...
}
```

Reply:
Yes I did it this way to test the builder setters, some of which such as `addFilter` have additional logic. I'll keep this as is out of convenience.